### PR TITLE
Fix: Handle linebreaks added to the executed command by the Telnet cl…

### PR DESCRIPTION
**Pull request by JJdeVries**
…ient

### CODE REVIEW

**Comment by kennedyshead** [comment]
I would love a unit-test for this, trying to get all the tests for connections up and running again ;)

File: aioasuswrt/connection.py [hunk 1]
```diff
@@ -143,6 +147,34 @@ async def async_connect(self):
                 b'#')).split(b'\n')[-1]
         self._connected = True
 
+        if self._linebreak is None:
+            await self._determine_linebreak()
+
+    async def _determine_linebreak(self):
+        """ Telnet or asyncio seems to be adding linebreaks due to terminal
+        size, try to determine here what the column number is."""
+        if not self.is_connected:
```
Comment (kennedyshead) [inline aioasuswrt/connection.py:156]: Not sure this is needed here

File: aioasuswrt/connection.py [hunk 2]
```diff
@@ -143,6 +147,34 @@ async def async_connect(self):
...
+            await self.async_connect()
+
+        # Let's send a long line, and read back how the reader splits the line
+        self._writer.write((" " * 200 + "\n").encode('ascii'))
+        data = (await self._reader.readuntil(self._prompt_string)
+                ).decode('utf-8').split('\n')
+
+        if len(data) == 1:
```
Comment (kennedyshead) [inline aioasuswrt/connection.py:164]: Id rather just this part will be the new function (aka this if/else)

**Comment by kennedyshead** [comment]
Nice one! How do you feel about a unit-test?

File: tests/test_connection.py [hunk 3]
```diff
@@ -220,50 +224,82 @@
...
+            input_bytes = (" " * i + "\n" + " " * 5).encode('ascii')
+            self.connection._determine_linelength(input_bytes)
+            self.assertEqual(self.connection._linebreak, i)
+
+            # And now with some more lines
+            input_bytes = ((" " * i + "\n") * 3 + " " * 5).encode('ascii')
+            self.connection._determine_linelength(input_bytes)
+            self.assertEqual(self.connection._linebreak, i)
+
+    @pytest.mark.skip(
```
Comment (kennedyshead) [inline tests/test_connection.py:258]: I think you should comment out all the ones below here, they are not ready to be readded

**Comment by JJdeVries** [comment]
Pfoe, so finally the tests are successful. For me this merge now seems done, the writer/reader mocks should enable a more robust checking of the TelnetConnection, although I think the most relevant parts are now captured.

**Pull request merged**